.github/workflows/ci.yml

```
@@ -57,56 +126,58 @@ jobs:
         with:
           python-version: "3.10"
 
-      - name: Build dev environment
+      - name: Login to GHCR
         run: |
-          ./dev_scripts/env.py --distro ${{ env.distro }} \
-              --version ${{ env.version }} \
-              build-dev
+          echo ${{ github.token }} | podman login ghcr.io -u USERNAME --password-stdin
 
-      - name: Install container build dependencies
-        run: sudo apt install pipx && pipx install poetry
+      - name: Get the dev environment
+        run: |
+          ./dev_scripts/env.py \
+              --distro ${{ matrix.distro }} \
+              --version ${{ matrix.version }} \
+              build-dev --sync
 
-      - name: Build Dangerzone image
-        run: python3 ./install/common/build-image.py
+      - name: Restore container cache
+        uses: actions/cache@v4
+        with:
+          key: v2-${{ hashFiles('Dockerfile', 'dangerzone/conversion/common.py', 'dangerzone/conversion/doc_to_pixels.py', 'dangerzone/conversion/pixels_to_pdf.py') }}
+          path: |-
+            share/container.tar.gz
+            share/image-id.txt
 
       - name: Build Dangerzone .deb
         run: |
-          ./dev_scripts/env.py --distro ${{ env.distro }} \
-              --version ${{ env.version }} \
+          ./dev_scripts/env.py --distro ${{ matrix.distro }} \
+              --version ${{ matrix.version }} \
               run --dev --no-gui ./dangerzone/install/linux/build-deb.py
 
       - name: Upload Dangerzone .deb
+        if: matrix.distro == 'debian' && matrix.version == 'trixie'
```

Nit, the reference distro where we build Debian packages is Debian bookworm:

```suggestion
        if: matrix.distro == 'debian' && matrix.version == 'bookworm'
```

---

Great! This is fixed in 9077dacdadc8f5614a8accfe74c193b8ec7be5e3